infra/src/main/java/com/redhat/service/bridge/infra/models/filters/FilterFactory.java

```diff
@@ -1,18 +1,31 @@
 package com.redhat.service.bridge.infra.models.filters;
 
+import java.io.IOException;
+import java.util.List;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+
 public class FilterFactory {
-    public static BaseFilter buildFilter(String type, String key, String value) {
-        switch (type) {
-            case StringBeginsWith.FILTER_TYPE_NAME:
-                return new StringBeginsWith(key, value);
-            case StringContains.FILTER_TYPE_NAME:
-                return new StringContains(key, value);
-            case StringEquals.FILTER_TYPE_NAME:
-                return new StringEquals(key, value);
-            case ValuesIn.FILTER_TYPE_NAME:
-                return new ValuesIn(key, value);
-            default:
-                throw new IllegalArgumentException("Type " + type + " is not valid");
+    public static BaseFilter<?> buildFilter(String type, String key, String value) {
+        try {
+            switch (type) {
+                case StringBeginsWith.FILTER_TYPE_NAME:
+                    return new StringBeginsWith(key,
+                            ObjectMapperFactory.get().readValue(value, new TypeReference<List<String>>() {
+                            }));
+                case StringContains.FILTER_TYPE_NAME:
+                    return new StringContains(key, ObjectMapperFactory.get().readValue(value, new TypeReference<List<String>>() {
+                    }));
+                case StringEquals.FILTER_TYPE_NAME:
+                    return new StringEquals(key, value);
+                case ValuesIn.FILTER_TYPE_NAME:
+                    return new ValuesIn(key, ObjectMapperFactory.get().readValue(value, new TypeReference<List<Object>>() {
+                    }));
+                default:
+                    throw new IllegalArgumentException("Type " + type + " is not valid");
```

Comment:
@fjtirado I think we should throw a subclass of (or `EventBridgeManagerException`). At this minute, this Exception will not be correctly handled by our Error mapping as that only handles `EventBridgeManagerException` instances.

Reply:
@robpblake After reviewing the usage of FilterFactory, this class is only used by tests, so Im now wondering if we sohuld just move it to test and completely forgot this discussion ;)

Reply:
Removing class and associated test